tests/test_package.py

```diff
@@ -59,45 +73,188 @@ def test_share_git_package():
+            torch_version, cuda_version
+        )
 
         assert (
-                formatted_install_cmd == expected_install_cmd
+            formatted_install_cmd == expected_install_cmd
         ), f"Unexpected result for {(torch_version, cuda_version)} "
 
 
+@pytest.mark.clustertest
+@pytest.mark.parametrize("cluster", ["v100", "k80", "a10g"], indirect=True)
+def test_getting_cuda_version_on_clusters(request, cluster):
+    """Gets the cuda version on the cluster and asserts it is the expected version"""
+    return_codes: list = cluster.run_python(
+        ["import runhouse as rh", "print(rh.Package.detect_cuda_version())"]
+    )
+    cuda_version_on_cluster = return_codes[0][1].strip().split("\n")[-1]
+    print(f"{cluster.name}: {cuda_version_on_cluster}")
+
+    instance_type = cluster.instance_type.lower()
+
+    # Save the cuda version for each cluster, to be used in the next test
+    request.config.cache.set(instance_type, cuda_version_on_cluster)
+
+    if instance_type.startswith("k80"):
+        assert cuda_version_on_cluster == "11.3"
+    else:
+        assert cuda_version_on_cluster == "11.7"
+
+
+@pytest.mark.clustertest
+@pytest.mark.parametrize("cluster", ["v100", "k80", "a10g"], indirect=True)
+def test_install_cmd_for_torch_on_cluster(request, cluster):
+    """Checks that the install command for torch runs properly on the cluster.
+    Confirms that we can properly install the package (and send a torch tensor to cuda to validate it"""
+    # Use the cuda version on the cluster that we got from the previous test
+    cuda_version_on_cluster = request.config.cache.get(
+        cluster.instance_type.lower(), None
+    )
+    assert cuda_version_on_cluster, (
+        f"No cuda version saved for {cluster.name} - run `test_getting_cuda_version_on_clusters`"
+        "to save the cuda version"
+    )
+
+    cuda_url = rh.Package.TORCH_INDEX_URLS_FOR_CUDA.get(cuda_version_on_cluster)
+    cuda_index_url = f"--index-url {cuda_url}"
+
+    install_commands_for_cluster = [
+        f"torch==1.13.1 {cuda_index_url} {extra_index_url}",
+        f"torch>=1.13.0 {cuda_index_url} {extra_index_url}",
+        f"torch>1.13.0 {cuda_index_url} {extra_index_url}",
+        f"torch {extra_index_url} {cuda_index_url}",
+        f"torch>=1.13.0,<=2.0.0 {cuda_index_url} {extra_index_url}",
+        f"torch==2.0.0 {cuda_index_url} {extra_index_url} torchaudio {cuda_index_url} {extra_index_url}",
+        f"torch~=2.0.0 {cuda_index_url} {extra_index_url}",
+    ]
+
+    for install_cmd in install_commands_for_cluster:
+        # Run the complete install command on the cluster
+        pip_install_cmd = cluster.run_python(
+            [
+                "import runhouse as rh",
+                f"rh.Package.pip_install(install_cmd='{install_cmd}')",
+            ]
+        )
+        assert (
+            "ERROR" not in pip_install_cmd[0][1]
+        ), f"Failed to install command on {cluster.name}"
```

Comment:
Don't we just want to do cluster.install_packages(["torch"]) for each of these to see if the installation works?